hs_core/models.py

```diff
@@ -3356,9 +3356,12 @@ def size(self):
 
         Raises SessionException if iRODS fails.
         """
+        # trigger file size read for files that haven't been set yet
+        for f in self.files.filter(_size__lt=0):
```

Comment:
This looks good.

Reply:
Now when I think about this for loop, it may create time out error (original problem) for any existing resource with a large number of files when that resource is accessed first time. Probably need to a management script to set file size.

Reply:
I pointed this out to David on Friday and said we could do a management script but mentioned it's probably not worth the effort because the first call is no slower than it is now.  He agreed.  I'm fine doing a script if someone really wants it, but I don't think it's necessary.

Reply:
OK.